doc/guides/dts/howtos.rst

```diff
@@ -204,21 +204,32 @@ variable :makevar:`DTC_OVERLAY_FILE` contains a space- or semicolon-separated
 list of overlays. If :makevar:`DTC_OVERLAY_FILE` specifies multiple files, they
 are included in that order by the C preprocessor.
 
-Here are some ways to set it:
-
-1. on the cmake build command line
-   (``-DDTC_OVERLAY_FILE="file1.overlay;file2.overlay"``)
-#. with the CMake ``set()`` command in the application ``CMakeLists.txt``,
-   before including zephyr's :file:`boilerplate.cmake` file
-#. create a ``boards/<BOARD>_<revision>.overlay`` file in the application
-   folder for the current board revision. This requires that the board supports
+Devicetree overlays can be specified by the user directly, in a
+:file:`CMakeLists.txt` file, or applied automatically.
+
+Overlay files are applied according to the following scheme:
+
+1. If ``DTC_OVERLAY_FILE`` is set, the overlay file(s) in it are merged and used
+   by the application. ``DTC_OVERLAY_FILE`` can be set in various ways:
```

Comment:
This is inaccurate or at best misleading and is better explained in the documentation already linked to at the top of the section.

This text implies that the overlays themselves are all that are merged together somehow. The truth is that BOARD.dts and any overlays are concatenated and the results are treated as the final devicetree after running the C preprocessor. So all the DTS files are concatenated; it's not that just the overlays are merged. (Note the confusion earlier in the discussion about DTC_OVERLAY_FILE vs. OVERLAY_CONFIG as evidence that this text is misleading.)

This is described in detail here: https://docs.zephyrproject.org/latest/guides/dts/intro.html#input-files. Note the link to the page containing this section at the top of this HOWTO.

Furthermore, this section is duplicating content already written in the introduction. This is the HOWTOs page; it's meant as a list of recipes, not a place to introduce new ideas. I don't think we should be mixing the two here.

Reply:
> This text implies that the overlays themselves are all that are merged together somehow. The truth is that BOARD.dts and any overlays are concatenated and the results are treated as the final devicetree after running the C preprocessor. 

True, internally they might be concatenated before running the preprocessor but the final DTS file will look merged.
That is, if I create an overlay for nrf52840dk as:
```
/ {
        leds {
                led0: led_0 {
                        gpios = < &gpio0 0xe 0x1 >;
                        label = "Green LED 0";
                };
        };
};
```
in order to change the pin that connects to the led, then the final zephyr.dts will contain:
```
...
      leds {
                compatible = "gpio-leds";
                led0: led_0 {
                        gpios = < &gpio0 0xe 0x1 >;
                        label = "Green LED 0";
                };
...
```
and not:
```
...
      leds {
                compatible = "gpio-leds";
                led0: led_0 {
                        gpios = < &gpio0 0xd 0x1 >;
                        label = "Green LED 0";
                };
...
/ {
        leds {
                led0: led_0 {
                        gpios = < &gpio0 0xe 0x1 >;
                        label = "Green LED 0";
                };
        };
};
```
and I think that is what the user cares about.

So I think merged is a better term than concatenated here, as concatenated could indicate they are just concatenated to form the final `zephyr.dts`.

Also merged is used here:
https://docs.zephyrproject.org/latest/guides/build/kconfig/setting.html#the-initial-configuration
> If CONF_FILE is set, the configuration file(s) specified in it are merged and used as the application configuration. CONF_FILE can be set in various ways:

where the files are processed in order and is updating the config setting to the value specified.
This means the last file specifying a given setting wins.

We could just as well have concatenated all conf files to a single file and passed that file to kconfig.py, and the last setting of a given value would win.
What is important is the final outcome `.config` / `zephyr.dts`, and there it could be seen as merged.


> Note the confusion earlier in the discussion about DTC_OVERLAY_FILE vs. OVERLAY_CONFIG as evidence that this text is misleading.

I don't see how a change from `merged` to `concatenate` makes it less confusing.
Conf files very much like device tree files whj.

For config files, the `<board>_defconfig` is not affected by `CONF_FILE` or `CONFIG_OVERLAY` settings.
Just as `<board>.dts` is not affected by the `DTS_OVERLAY_FILE`

I don't see how any re-wording here can remove that confusion to users.
For conf files we have:
- `<board>_defconfig` concatenated with `CONF_FILES` and `OVERLAY_CONFIG` to form `.config` from Kconfig tree.
  - Setting `CONF_FILES` will disable automatically selected `CONF_FILES` such as `prj.conf` and `boards/<board>.conf`
  - Setting `OVERLAY_CONFIG` will impact automatic selected `CONF_FILES`

  All files are still concatenated with `<board>_defconfig`
- `<board>.dts` concatenated with `DTS_OVERLAY_FILE`
  - Setting `DTS_OVERLAY_FILE` will disable automatically selected `DTS_OVERLAY_FILES` such as `app.dts` or `<board>.dts`

So the word `OVERLAY` has one behavior for conf files and another for devicetree.

> This is described in detail here: https://docs.zephyrproject.org/latest/guides/dts/intro.html#input-files. Note the link to the page containing this section at the top of this HOWTO.

And notice that link is inside a `tip` note describing how a `base` devicetree can be obtained.
Such link should probably be more generic link to inform the user of the different file types.
There is of course also the [BOARD.dts](https://docs.zephyrproject.org/latest/guides/dts/intro.html#devicetree-in-out-files) link, but it's not as clear to the reader that points to overlay files.

> This is the HOWTOs page; it's meant as a list of recipes, not a place to introduce new ideas. I don't think we should be mixing the two here.

i'm not changing the purpose of what is already here, just trying to clear up the existing text.
But I would agree most of this section probably belong better in the `intro` than the `howto`, so if you prefer I will move the text.

Reply:
You're missing my point.

I am objecting to language saying "the overlays are merged".

I object because that omits that BOARD.dts is part of what is being merged.